dali/operators/signal/decibel/to_decibels_op_gpu.cu

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <utility>
+#include <vector>
+#include <memory>
+#include "dali/operators/signal/decibel/to_decibels_op.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/kernel_params.h"
+#include "dali/kernels/reduce/reduce_all_kernel_gpu.h"
+#include "dali/kernels/signal/decibel/to_decibels_gpu.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+template <typename T, int Dims>
+class ToDecibelsImpl : public OpImplBase<GPUBackend> {
+ public:
+  using MaxKernel = kernels::reduce::ReduceAllGPU<T, T, Dims, kernels::reductions::max>;
+  using ToDecibelsKernel = kernels::signal::ToDecibelsGpu<T, Dims>;
+  using ToDecibelsArgs = kernels::signal::ToDecibelsArgs<T>;
+
+  explicit ToDecibelsImpl(ToDecibelsArgs args)
+      : args_(std::move(args)) {
+    if (args_.ref_max) {
+      kmgr_max_.Initialize<MaxKernel>();
+      kmgr_max_.Resize<MaxKernel>(1, 1);
+    }
+    kmgr_todb_.Initialize<ToDecibelsKernel>();
+    kmgr_todb_.Resize<ToDecibelsKernel>(1, 1);
+  }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<GPUBackend> &ws) override;
+  void RunImpl(workspace_t<GPUBackend> &ws) override;
+
+ private:
+  ToDecibelsArgs args_;
+
+  kernels::KernelManager kmgr_max_;
+  kernels::KernelManager kmgr_todb_;
+
+  std::vector<OutputDesc> max_out_desc_;
+  TensorList<GPUBackend> max_out_;
+};
+
+template <typename T, int Dims>
+bool ToDecibelsImpl<T, Dims>::SetupImpl(std::vector<OutputDesc> &output_desc,
```

Comment:
It's something I've missed in the kernel review: do we need Dims as a compile time argument at all? If we flatten the tensor in the kernel anyway, it can just as well be DynamicDimensions, which would save us some compile time.

Reply:
Sounds good